src/emulator/storage/rules/manager.ts

```diff
@@ -100,13 +120,50 @@ export class StorageRulesManager {
   }
 }
 
-function readSourceFile(fileName: string): string {
-  try {
-    return fs.readFileSync(fileName).toString();
-  } catch (error: any) {
-    if (error.code === "ENOENT") {
-      throw new FirebaseError(`File not found: ${fileName}`);
+/**
+ * Maintains a mapping from storage resource to {@link StorageRulesManagerImplementation} and
+ * directs calls to the appropriate instance.
+ */
+class StorageRulesManagerRegistry implements StorageRulesManager {
+  private _rulesManagers: Map<string, StorageRulesManagerImplementation>;
+
+  constructor(_initRules: RulesConfig[], private _runtime: StorageRulesRuntime) {
+    this._rulesManagers = new Map<string, StorageRulesManagerImplementation>();
+    for (const { resource, rules } of _initRules) {
+      this.createRulesManager(resource, rules);
     }
-    throw error;
+  }
+
+  async start(): Promise<StorageRulesIssues> {
+    const allIssues = new StorageRulesIssues();
+    for (const rulesManager of this._rulesManagers.values()) {
+      allIssues.extend(await rulesManager.start());
+    }
+    return allIssues;
+  }
+
+  getRuleset(resource: string): StorageRulesetInstance | undefined {
+    return this._rulesManagers.get(resource)?.getRuleset();
+  }
+
+  async setSourceFile(rules: SourceFile, resource: string): Promise<StorageRulesIssues> {
+    const rulesManager =
+      this._rulesManagers.get(resource) || this.createRulesManager(resource, rules);
+    return rulesManager.setSourceFile(rules);
+  }
+
+  async close(): Promise<void> {
+    for (const rulesManager of this._rulesManagers.values()) {
+      await rulesManager.close();
```

Comment:
You can speed this up by using `await Promise.all(...)`

https://www.codecademy.com/courses/asynchronous-javascript/lessons/async-await/exercises/await-promise-all

Reply:
Done.